connector.rb

```diff
@@ -1,37 +1,2798 @@
+                  'https://oauth.4me.com/token'
+                end
+              when 'quality_assurance'
+                case region
+                when 'au'
+                  'https://oauth.au.4me.qa/token'
+                when 'uk'
+                  'https://oauth.uk.4me.qa/token'
+                when 'us'
+                  'https://oauth.us.4me.qa/token'
+                when 'ch'
+                  'https://oauth.ch.4me.qa/token'
+                else
+                  'https://oauth.4me.qa/token'
+                end
+              when 'demo'
+                'https://oauth.4me-demo.com/token'
+              else
+                "https://oauth.#{custom_domain_name}/token"
+              end
+
+            request = post(token_url)
+            request.headers('x-4me-Account': connection['account'])
+            payload = {
+              grant_type: 'client_credentials'
+            }.compact
+            payload = payload.merge({
+              client_id: connection['client_id'],
+              client_secret: connection['client_secret']
+            })
+            request = request.payload(payload)
+            request.request_format_www_form_urlencoded
+          end,
+
+          apply: lambda do |connection|
+            if current_url.include?('https://graphql.')
+              token_type = connection['token_type'] || 'Bearer'
+              access_token = connection['access_token']
+
+              headers(Authorization: "#{token_type} #{access_token}") unless access_token.blank?
+            end
+          end,
+
+          refresh_on: [401, 403]
+        }
+      }
     },
 
     base_uri: lambda do |connection|
-      ''
+      instance = connection['instance']
+      region = connection['region']
+      custom_domain_name = connection['custom_domain_name']
+
+      case instance
+      when 'production'
+        case region
+        when 'au'
+          'https://graphql.au.4me.com'
+        when 'uk'
+          'https://graphql.uk.4me.com'
+        when 'us'
+          'https://graphql.us.4me.com'
+        when 'ch'
+          'https://graphql.ch.4me.com'
+        else
+          'https://graphql.4me.com'
+        end
+      when 'quality_assurance'
+        case region
+        when 'au'
+          'https://graphql.au.4me.qa'
+        when 'uk'
+          'https://graphql.uk.4me.qa'
+        when 'us'
+          'https://graphql.us.4me.qa'
+        when 'ch'
+          'https://graphql.ch.4me.qa'
+        else
+          'https://graphql.4me.qa'
+        end
+      when 'demo'
+        'https://graphql.4me-demo.com'
+      else
+        "https://graphql.#{custom_domain_name}"
+      end
     end
   },
 
+  # Tests the connection to the server by performing
+  # a simple introspection query.
   test: lambda do |connection|
+    call(
+      'run_gql',
+      connection,
+      "{
+        __schema {
+          queryType {
+            name
+          }
+        }
+      }",
+      nil,
+      connection['account']
```

Comment:
Why we cannot get account from connection inside the function? Hence we have can reduce number of params for `run_gql` method.

Reply:
The default account value is the one from the connection, but in every action, I can specify another account. The authorization can grant me access to multiple 4me accounts, which is why the param is required in the `run_gql`.